rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -90,23 +91,69 @@ impl ArrowBlockfileWriter {
         }
     }
 
-    pub(crate) fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
-        self,
+    pub(crate) async fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
+        mut self,
     ) -> Result<ArrowBlockfileFlusher, Box<dyn ChromaError>> {
         let mut blocks = Vec::new();
+        let mut handled_blocks = HashSet::new();
         for (_, delta) in self.block_deltas.lock().drain() {
+            handled_blocks.insert(delta.id);
```

Comment:
this hashset is used at most only once in the entire lifetime of the system but it is set every time commit runs. Can it also be set only once?

Reply:
I don’t understand. This hashset only exists inside of commit, and commit takes ownership of self. You can only call commit () once per a block files lifetime since it take’s ownership

Reply:
oh I meant something simple like:

`if self.root.version == Version::V1 {
   handled_blocks.insert(delta.id);
}`

to avoid unnecessary insertions to this hashset when it is not used

Reply:
I don't think that its worth the further code pollution and option / null handling